test/mocha/10-restrictions.js

```diff
@@ -311,61 +345,99 @@ describe('restrictions', function() {
   });
 
   it('should remove a restriction from the database', async function() {
-    // create restriction
-    const id = await generateId();
-    const actualRestriction = await restrictions.insert({
-      restriction: {
-        id,
-        zone: ZONES.ONE,
-        resource: RESOURCES.MANGO,
-        method: 'limitOverDuration',
-        methodOptions: {
-          limit: 1,
-          duration: 'P30D'
-        }
+    // create restrictions
+    const mockRestriction1 = {
+      id: await generateId(),
+      zone: ZONES.ONE,
+      resource: RESOURCES.ASPARAGUS,
+      method: 'limitOverDuration',
+      methodOptions: {
+        limit: 10,
+        duration: 'P30D'
       }
-    });
-    const expectedRestriction = {
-      id,
+    };
+    const mockRestriction2 = {
+      id: await generateId(),
       zone: ZONES.ONE,
-      resource: RESOURCES.MANGO,
+      resource: RESOURCES.ASPARAGUS,
       method: 'limitOverDuration',
       methodOptions: {
         limit: 1,
-        duration: 'P30D'
+        duration: 'P7D'
       }
     };
-    should.exist(actualRestriction);
-    should.exist(actualRestriction.meta);
-    should.exist(actualRestriction.restriction);
-    actualRestriction.restriction.should.deep.equal(expectedRestriction);
-
-    const getRestriction = await restrictions.get({
+    await restrictions.insert({
+      restriction: mockRestriction1
+    });
+    await restrictions.insert({
+      restriction: mockRestriction2
+    });
+    const restrictionsArray = await restrictions.getAll({
       zone: ZONES.ONE,
-      resource: RESOURCES.MANGO
+      resource: RESOURCES.ASPARAGUS
     });
-    should.exist(getRestriction);
-    should.exist(getRestriction.meta);
-    should.exist(getRestriction.restriction);
-    getRestriction.restriction.should.deep.equal(expectedRestriction);
+    should.exist(restrictionsArray);
+    restrictionsArray.restrictions.length.should.equal(2);
+    restrictionsArray.restrictions[0].restriction.should.eql(mockRestriction1);
+    restrictionsArray.restrictions[1].restriction.should.eql(mockRestriction2);
```

Comment:
Is there a sort that guarantees that these will come out in any particular order?  I think we're relying on the natural order, or the order of insertion here.  If it works, I would suggest using the same `deep.members` construction I suggested elsewhere.  If we don't build in expectations about order into the API now, we will be able to potentially add sorting later as a non-breaking feature release.

Reply:
Understood, however since we are returning these from the database, there is still the `meta` property attached, which does not work with the `should.have.deep.members`. Is there a way I can still test for the `restriction` property without the `meta`?

Reply:
You could, for now, just run `.map` and map the records to just restrictions and check those. That won't give us full coverage on checking everything (as we're intentionally not checking meta, etc.), but would be ok for now, IMO.